File: app/src/main/java/com/erif/contentloader/example/ActivityBanner.java
package com.erif.contentloader.example;

import static java.lang.Math.abs;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.erif.contentloader.LoaderContainer;
import com.erif.contentloader.R;
import com.erif.contentloader.helper.timer.Delay;
import com.google.android.material.imageview.ShapeableImageView;

public class ActivityBanner extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_banner);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("Banner / Slider Loader");
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        LoaderContainer loader = findViewById(R.id.act_frame_loader);
        RelativeLayout image = findViewById(R.id.act_frame_layoutImage);
        loader.startAndHideContent(image, true);

        new Delay(2.5, () -> loader.stopAndShowContent(image, true)).start();

        int sWidth = getResources().getDisplayMetrics().widthPixels;
        ShapeableImageView imgLeft = findViewById(R.id.act_frame_imgLeft);
        ViewGroup.MarginLayoutParams paramLeft = (ViewGroup.MarginLayoutParams) imgLeft.getLayoutParams();
        float snipped = sWidth * -1;
        float transX = snipped + (paramLeft.getMarginEnd() * 2);
        imgLeft.setTranslationX(transX);

        ShapeableImageView imgRight = findViewById(R.id.act_frame_imgRight);
        imgRight.setTranslationX(abs(transX));

        ShapeableImageView imgCenter = findViewById(R.id.act_frame_imgCenter);
        ViewGroup.MarginLayoutParams paramCenter = (ViewGroup.MarginLayoutParams) imgCenter.getLayoutParams();
        paramCenter.setMarginStart(paramCenter.getMarginStart() * 2);
        paramCenter.setMarginEnd(paramCenter.getMarginEnd() * 2);
        imgCenter.setLayoutParams(paramCenter);

    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/erif/contentloaderdemo/example/ActivityBanner.kt
package com.erif.contentloaderdemo.example

import android.os.Bundle
import android.view.MenuItem
import android.view.ViewGroup.MarginLayoutParams
import android.widget.RelativeLayout
import androidx.appcompat.app.AppCompatActivity
import com.erif.contentloader.LoaderContainer
import com.erif.contentloaderdemo.R
import com.erif.contentloaderdemo.helper.timer.Delay
import com.google.android.material.imageview.ShapeableImageView
import kotlin.math.abs

class ActivityBanner: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_banner)
        supportActionBar?.let {
            it.title = "Banner / Slider Loader"
            it.setDisplayHomeAsUpEnabled(true)
        }

        val loader = findViewById<LoaderContainer>(R.id.act_frame_loader)
        val image = findViewById<RelativeLayout>(R.id.act_frame_layoutImage)
        loader.startAndHideContent(image, true)

        Delay(
            2.5
        ) { loader.stopAndShowContent(image, true) }.start()

        val sWidth = resources.displayMetrics.widthPixels
        val imgLeft = findViewById<ShapeableImageView>(R.id.act_frame_imgLeft)
        val paramLeft = imgLeft.layoutParams as MarginLayoutParams
        val snipped = (sWidth * -1).toFloat()
        val transX = snipped + paramLeft.marginEnd * 2
        imgLeft.translationX = transX

        val imgRight = findViewById<ShapeableImageView>(R.id.act_frame_imgRight)
        imgRight.translationX = abs(transX)

        val imgCenter = findViewById<ShapeableImageView>(R.id.act_frame_imgCenter)
        val paramCenter = imgCenter.layoutParams as MarginLayoutParams
        paramCenter.marginStart = paramCenter.marginStart * 2
        paramCenter.marginEnd = paramCenter.marginEnd * 2
        imgCenter.layoutParams = paramCenter
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

}